src/Angor/Client/Pages/Settings.razor

```diff
@@ -471,39 +472,81 @@
             await Refresh(false);
         }
     }
-
+    
     private async Task AddIndexer()
     {
         if (!string.IsNullOrWhiteSpace(newIndexerLink))
         {
             if (!Uri.TryCreate(newIndexerLink, UriKind.Absolute, out Uri? uri))
             {
-                notificationComponent.ShowErrorMessage($"invalid url {newIndexerLink}");
+                notificationComponent.ShowErrorMessage($"Invalid URL: {newIndexerLink}");
                 return;
             }
 
             if (uri.Scheme is not ("http" or "https"))
             {
-                notificationComponent.ShowErrorMessage($"invalid url {newIndexerLink} schema must be http or https");
+                notificationComponent.ShowErrorMessage($"Invalid URL schema. Must be HTTP or HTTPS.");
                 return;
             }
 
-            newIndexerLink = new Uri($"{uri.Scheme}://{uri.Host}").AbsoluteUri.TrimEnd('/');
+            newIndexerLink = $"{uri.Scheme}://{uri.Host}".TrimEnd('/');
+
+            Logger.LogInformation($"Normalized URL: {newIndexerLink}");
 
             if (settingsInfo.Indexers.Any(a => a.Url == newIndexerLink))
             {
-                notificationComponent.ShowErrorMessage($"url exists {newIndexerLink}");
+                notificationComponent.ShowErrorMessage($"URL already exists: {newIndexerLink}");
                 return;
             }
 
-            settingsInfo.Indexers.Add(new SettingsUrl { Url = newIndexerLink, IsPrimary = !settingsInfo.Indexers.Any() });
+            // Add the new indexer temporarily
+            settingsInfo.Indexers.Add(new SettingsUrl
+            {
+                Url = newIndexerLink,
+                LastCheck = DateTime.MinValue, 
+                Status = UrlStatus.NotReady
+            });
+
             _clientStorage.SetSettingsInfo(settingsInfo);
+
+            await _networkService.CheckServices(force: true);
+
+            settingsInfo = _clientStorage.GetSettingsInfo();
+
+            // Check if the newly added indexer is online
+            var addedIndexer = settingsInfo.Indexers.FirstOrDefault(i => i.Url == newIndexerLink);
+            if (addedIndexer?.Status != UrlStatus.Online)
+            {
+                notificationComponent.ShowErrorMessage("The indexer is offline or unreachable.");
+                // Remove the indexer if it's not online
+                settingsInfo.Indexers.Remove(addedIndexer);
+                _clientStorage.SetSettingsInfo(settingsInfo);
+                return;
+            }
+
+            // Validate genesis block hash if the indexer is online
+            var fetchedHash = await _indexerService.GetGenesisBlockHashAsync(newIndexerLink);
+            var expectedHash = _networkConfiguration.GetGenesisBlockHash();
+
+            if (string.IsNullOrEmpty(fetchedHash) || !_indexerService.ValidateGenesisBlockHash(fetchedHash, expectedHash))
```

Comment:
just do this check `string.IsNullOrEmpty` inside the method `ValidateGenesisBlockHash`